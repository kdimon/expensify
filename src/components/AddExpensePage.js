import React from 'react';

const AddExpensePage = () => (
    <div>
        This is from create
    </div>
);

export default AddExpensePage;