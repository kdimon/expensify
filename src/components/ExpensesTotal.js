import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';

export class ExpensesTotal extends React.Component {
    constructor() {
        super();
        this.getExpensesTotal = this.getExpensesTotal.bind(this);
    }

    getExpensesTotal(expenses) {
        if(expenses.length === 1) {
            return expenses[0].amount;
        }else {
            return expenses
                .map(expense => {
                    return expense.amount;
                })
                .reduce((total, amount) => {
                    return total + amount;
                })
            ;
        }
    }

    render() {
        return (
            <div>
                {this.props.expenses.length === 0 ? <p>No expenses</p> : <p>Total expenses: {numeral(this.getExpensesTotal(this.props.expenses) / 100).format('$0,0.00')}</p>}
            </div>
        );
    }
}

const MapStateToProps = state => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(MapStateToProps)(ExpensesTotal);