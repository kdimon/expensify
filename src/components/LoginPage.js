import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from "../actions/auth";

export const LoginPage = ({startLogin}) => (
    <div className='box-layout'>
        <div className='box-layout__box'>
            <h1 className='box-layout__title'>Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className='button button--login' id='google' onClick={startLogin}>Log in with Google <img className='social-icon' src="/images/icons/gmail.svg" /></button>
            <button className='button button--login' id='facebook' onClick={startLogin}>Log in with Facebook <img className='social-icon' src="/images/icons/facebook.svg" /></button>
            <button className='button button--login' id='github' onClick={startLogin}>Log in with GitHub <img className='social-icon' src="/images/icons/github.svg" /></button>
            <button className='button button--login' id='twitter' onClick={startLogin}>Log in with Twitter <img className='social-icon' src="/images/icons/twitter.svg" /></button>
        </div>
    </div>
);

const mapDispatchToProps = dispatch => ({
    startLogin: e => dispatch(startLogin(e))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);