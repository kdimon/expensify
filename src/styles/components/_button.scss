.button {
    background-color: $blue;
    border: none;
    color: $white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    outline: none;
    padding: $s-size;
    text-decoration: none;

    &--link {
        background-color: transparent;
    }

    &--secondary {
        background-color: $light-grey__button;
    }

    &--login {
        align-items: center;
        border-radius: 3px;
        display: flex;
        justify-content: space-between;
        width: 100%;

        &:not(:last-child) {
            margin-bottom: $s-size;
        }

        &__google {
            background-color: $red-google;
        }

        &__facebook {
            background-color: $blue-facebook;
        }

        &__github {
            background-color: $blue-github;
        }

        &__twitter {
            background-color: $blue-twitter;
        }

        .social-icon {
            height: 24px;
            width: 24px;
        }
    }
}