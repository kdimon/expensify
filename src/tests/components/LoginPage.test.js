import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';

let startLogin,
    wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin} />);
});

test('should correctly render LoginPage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on google button click', () => {
    wrapper.find('#google').simulate('click');

    expect(startLogin).toHaveBeenCalled();
});

test('should call startLogin on facebook button click', () => {
    wrapper.find('#facebook').simulate('click');

    expect(startLogin).toHaveBeenCalled();
});

test('should call startLogin on github button click', () => {
    wrapper.find('#github').simulate('click');

    expect(startLogin).toHaveBeenCalled();
});

test('should call startLogin on twitter button click', () => {
    wrapper.find('#twitter').simulate('click');

    expect(startLogin).toHaveBeenCalled();
});