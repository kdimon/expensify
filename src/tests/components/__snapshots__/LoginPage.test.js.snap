// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <button
      className="button button--login button--login__google"
      id="google"
      onClick={[Function]}
    >
      Log in with Google 
      <img
        className="social-icon"
        src="/images/icons/gmail.svg"
      />
    </button>
    <button
      className="button button--login button--login__facebook"
      id="facebook"
      onClick={[Function]}
    >
      Log in with Facebook 
      <img
        className="social-icon"
        src="/images/icons/facebook.svg"
      />
    </button>
    <button
      className="button button--login button--login__github"
      id="github"
      onClick={[Function]}
    >
      Log in with GitHub 
      <img
        className="social-icon"
        src="/images/icons/github.svg"
      />
    </button>
    <button
      className="button button--login button--login__twitter"
      id="twitter"
      onClick={[Function]}
    >
      Log in with Twitter 
      <img
        className="social-icon"
        src="/images/icons/twitter.svg"
      />
    </button>
  </div>
</div>
`;
